code/dynamic_programming/src/LongestRepeatingSubsequence.cpp

```diff
@@ -0,0 +1,56 @@
+//dynamic programming || Longest repeating subsequence
+
+#include <iostream>
+#include<cstring>
+using namespace std;
+
+int longest_repeating_subsequence(string s)
+{
+    int n=s.size();  // Obtaining the length of the string
+    int dp[n+1][n+1];

```

Comment:
Biggest issue I have with this is that it makes use of VLAs, which are not part of standard C++. Here are better suggestions:

- Nested [`std::vector`](http://en.cppreference.com/w/cpp/container/vector) (Might be overkill for such a small array or big array depending on what is passed)?
- Dynamic nested array (Will be painful to manage)
- [`std::unique_ptr`](http://en.cppreference.com/w/cpp/memory/unique_ptr) to nested dynamic arrays. However, while this solves the problems, it is somewhat unusual.

Pick one of the three options above. Or figure out something better.